### PR TITLE
Refactor papers fetching from ORCiD profile

### PR DESCRIPTION
**Pull request by RaitoBezarius**
From a discussion with @wetneb over IRC, we agreed that the function was really too complex and difficult to make sense of it.

I propose this change which attempts to improve drastically readability of the process.

I would like to unify the creation of paper under the `DataPaper` interface, but I need more time to figure out if it is really a good idea or not.

For now, I think that it's fair enough to move on, and fix some "critical" bugs.

### CODE REVIEW

File: backend/orcid.py [hunk 1]
```diff
@@ -144,133 +345,41 @@ def fetch_orcid_records(self, id, profile=None, use_doi=True):
...
-                            source=orcid_oai_source,
-                            identifier='orcid:'+id+':'+metadata['DOI'],
-                            splash_url='http://orcid.org/'+id,
-                            pubtype=paper.doctype)
-                    paper.add_oairecord(record)
-                    yield paper
-                except (KeyError, ValueError, TypeError):
-                    pass
-
-orcid_oai_source, _ = OaiSource.objects.get_or_create(identifier='orcid',
```
Comment (wetneb) [inline backend/orcid.py]: This should not be deleted! That's a global line of code (lying outside the previous function). It's also my fault, I should have put this at the beginning of the file instead.

Comment (RaitoBezarius) [inline backend/orcid.py]: I've put it at the beginning of the file

File: backend/orcid.py [hunk 2]
```diff
@@ -76,6 +75,10 @@
         'data-set':'dataset',
     }
 
+orcid_oai_source, _ = OaiSource.objects.get_or_create(identifier='orcid',
```
Comment (RaitoBezarius) [inline backend/orcid.py]: @wetneb No worries, it is still here :D.

Comment (wetneb) [inline backend/orcid.py]: Haha, sorry, thanks!

**Pull request merged**